x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformState.java

```diff
@@ -195,7 +212,11 @@ public String getWriteableName() {
     public void writeTo(StreamOutput out) throws IOException {
         taskState.writeTo(out);
         indexerState.writeTo(out);
-        out.writeMap(currentPosition);
+        if (out.getVersion().onOrAfter(Version.V_7_3_0)) {
```

Comment:
The corresponding constructor is using `Version.CURRENT`.  If this is tested anywhere then that mismatch will cause the CI to fail.

Reply:
i have set proper versions now and disabled bwc, to be re-enabled once backports are done